tests/console/btrfs_check.pm

```diff
@@ -0,0 +1,47 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: btrfs_check
+# Summary: Test that btrfs is setup properly
+# Maintainer: QE-C team <qa-c@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+sub run {
+    select_console 'root-console';
+
+    my $kiwi_volumes = qq(
+/
+/.snapshots
+/boot/grub2/i386-pc
+/boot/grub2/x86_64-efi
+/home
+/opt
+/root
+/srv
+/tmp
+/usr/local
+/var);
+
+    my $test_volumes = script_output("lsblk --list --noheadings --output MOUNTPOINTS /dev/vda3 | sort");
+    record_info('test_volumes', "$test_volumes");
+    record_info('kiwi_volumes', "$kiwi_volumes");
+    die "Volumes mismatch from Kiwi definition"
+      if $kiwi_volumes !~ $test_volumes;
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    assert_script_run('rpm -qa > /tmp/rpm_qa.txt');
+    upload_logs('/tmp/rpm_qa.txt');
+    upload_logs('/var/log/zypper.log');
+    $self->SUPER::post_fail_hook;
```

Comment:
This needs to be reworked. We are interested in the filesystem, hence a custom post_fail_hook would suit better here

Reply:
This is my first time writing a `post_fail_hook`, so I just copy-pasted :sweat_smile:

Reply:
IMO this does not need a custom `post_fail_hook` at all. All relevant info for possible failures is already visible.